data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusSinkService.java

```diff
@@ -4,49 +4,415 @@
+            prometheusSinkConfiguration.validateAndInitializeCertAndKeyFileInS3();
+            this.httpClientConnectionManager = new HttpClientSSLConnectionManager()
+                    .createHttpClientConnectionManager(prometheusSinkConfiguration, certificateProviderFactory);
+        }
+        this.httpAuthOptions = buildAuthHttpSinkObjectsByConfig(prometheusSinkConfiguration);
     }
 
     /**
-     * This method process buffer records and send to Prometheus End points based on configured codec
+     * This method process buffer records and send to Http End points based on configured codec
      * @param records Collection of Event
      */
     public void output(Collection<Record<Event>> records) {
         reentrantLock.lock();
         try {
             records.forEach(record -> {
                 final Event event = record.getData();
+                byte[] bytes = null;
+                if (event.getMetadata().getEventType().equals("METRIC")) {
+                    Remote.WriteRequest message = null;
+                    if (event instanceof JacksonGauge) {
+                        JacksonGauge jacksonGauge = (JacksonGauge) event;
+                        message = buildRemoteWriteRequest(jacksonGauge.getTime(),
+                                jacksonGauge.getStartTime(), jacksonGauge.getValue(), jacksonGauge.getAttributes(),jacksonGauge.getName());
+                    } else if (event instanceof JacksonSum) {
+                        JacksonSum jacksonSum = (JacksonSum) event;
+                        message = buildRemoteWriteRequest(jacksonSum.getTime(),
+                                jacksonSum.getStartTime(), jacksonSum.getValue(), jacksonSum.getAttributes(), jacksonSum.getName());
+                    } else if (event instanceof JacksonSummary) {
+                        JacksonSummary jacksonSummary = (JacksonSummary) event;
+                        message = buildRemoteWriteRequest(jacksonSummary.getTime(),
+                                jacksonSummary.getStartTime(), jacksonSummary.getSum(), jacksonSummary.getAttributes(), jacksonSummary.getName());
+                    } else if (event instanceof JacksonHistogram) {
+                        JacksonHistogram jacksonHistogram = (JacksonHistogram) event;
+                        message = buildRemoteWriteRequest(jacksonHistogram.getTime(),
+                                jacksonHistogram.getStartTime(), jacksonHistogram.getSum(), jacksonHistogram.getAttributes(), jacksonHistogram.getName());
+                    } else if (event instanceof JacksonExponentialHistogram) {
+                        JacksonExponentialHistogram jacksonExpHistogram = (JacksonExponentialHistogram) event;
+                        message = buildRemoteWriteRequest(jacksonExpHistogram.getTime(),
+                                jacksonExpHistogram.getStartTime(), jacksonExpHistogram.getSum(), jacksonExpHistogram.getAttributes(), jacksonExpHistogram.getName());
+                    } else {
```

Comment:
I see `getSum` method is not part of interface. Can we avoid this multiple if else statement and use reflection to call the `getSum` method ?

Reply:
@dinujoh Thanks for your review. Will update the code accordingly.

Reply:
@dinujoh Can we please elaborate on that?